test/test_railwaynetwork.cpp

```diff
@@ -1,1085 +1,2469 @@
-    network.change_edge_property(0, 2, "length");
-    EXPECT_TRUE(network.get_edge(0).length == 2);
-    network.change_edge_property(0, 3, "max_speed");
-    EXPECT_TRUE(network.get_edge(0).max_speed == 3);
-    network.change_edge_property(0, 4, "min_block_length");
-    EXPECT_TRUE(network.get_edge(0).min_block_length == 4);
-    network.change_edge_property(0, 1, 5, "length");
-    EXPECT_TRUE(network.get_edge(0).length == 5);
-    network.change_edge_property(0, 1, 6, "max_speed");
-    EXPECT_TRUE(network.get_edge(0).max_speed == 6);
-    network.change_edge_property(0, 1, 7, "min_block_length");
-    EXPECT_TRUE(network.get_edge(0).min_block_length == 7);
-    network.change_edge_property("v0", "v1", 8, "length");
-    EXPECT_TRUE(network.get_edge(0).length == 8);
-    network.change_edge_property("v0", "v1", 9, "max_speed");
-    EXPECT_TRUE(network.get_edge(0).max_speed == 9);
-    network.change_edge_property("v0", "v1", 10, "min_block_length");
-    EXPECT_TRUE(network.get_edge(0).min_block_length == 10);
-    network.change_edge_breakable(1, true);
-    EXPECT_TRUE(network.get_edge(1).breakable);
-    network.change_edge_breakable(1, 2, false);
-    EXPECT_FALSE(network.get_edge(1).breakable);
-    network.change_edge_breakable("v1", "v2", true);
-    EXPECT_TRUE(network.get_edge(1).breakable);
-
-    // out and in edges tests
-    std::vector<int> expected_out {1,2};
-    std::vector<int> expected_in {0};
-    std::vector<int> expected_neighbors {0,2};
-    auto out_edges_1 = network.out_edges(1);
-    std::sort(out_edges_1.begin(), out_edges_1.end());
-    EXPECT_TRUE(out_edges_1 == expected_out);
-    auto out_edges_v1 = network.out_edges("v1");
-    std::sort(out_edges_v1.begin(), out_edges_v1.end());
-    EXPECT_TRUE(out_edges_v1 == expected_out);
-    auto in_edges_1 = network.in_edges(1);
-    std::sort(in_edges_1.begin(), in_edges_1.end());
-    EXPECT_TRUE(in_edges_1 == expected_in);
-    auto in_edges_v1 = network.in_edges("v1");
-    std::sort(in_edges_v1.begin(), in_edges_v1.end());
-    EXPECT_TRUE(in_edges_v1 == expected_in);
-    auto neighbors_1 = network.neighbors(1);
-    std::sort(neighbors_1.begin(), neighbors_1.end());
-    EXPECT_TRUE(neighbors_1 == expected_neighbors);
-    auto neighbors_v1 = network.neighbors("v1");
-    std::sort(neighbors_v1.begin(), neighbors_v1.end());
-    EXPECT_TRUE(neighbors_v1 == expected_neighbors);
-
-    // successor tests
-    std::vector<int> expected_successors {1};
-    EXPECT_TRUE(network.get_successors(0) == expected_successors);
-    EXPECT_TRUE(network.get_successors(0, 1) == expected_successors);
-    EXPECT_TRUE(network.get_successors("v0", "v1") == expected_successors);
-
-    // Vertex and edge numbers
-    EXPECT_TRUE(network.number_of_vertices() == 3);
-    EXPECT_TRUE(network.number_of_edges() == 4);
-
-    // Valid successor
-    EXPECT_TRUE(network.is_valid_successor(0, 1));
-    EXPECT_FALSE(network.is_valid_successor(0, 2));
+  cda_rail::Network network;
```

Comment:
I think this graph is created more than once? GoogleTest allows for Setup and Teardown functionality as in other unit test frameworks. Consider using these to avoid multiple constructions of the same/similar graphs.

Reply:
Thank you. I will have a look into this, maybe in a separate PR.